### ISSUE TITLE
Passkey support incomplete on MacOS 10.9.5

### ISSUE THREAD

**Issue by Stecca6663** [i3roly/firefox-dynasty#24]
First of all, thank you for fixing the microphone issue and sorry for creating a new one so soon. I noticed that supposedly passkeys work  but only with a security key, but Firefox on modern macOS and chromium legacy on 10.9.5 offer support for the QR code version, so that you can use your mobile device to log in. Though on Chromium Legacy my IPhone gets stuck on "connecting", even though Bluetooth and Wi-Fi are enabled on the MacBook, so it might be a bigger issue, not sure. Here's some screenshots for context (First is Firefox, second is chromium legacy). Thank you as always!
![Screen Shot 2024-12-08 at 10 42 20 AM](https://github.com/user-attachments/assets/010c944b-4c96-4ae1-bbea-ff431b8b7bc9)
![Screen Shot 2024-12-08 at 10 41 49 AM](https://github.com/user-attachments/assets/a3ce3e41-4c3a-4168-9309-6723f6bb419d)

**Comment by i3roly** [comment]
even on my macbook pro that runs crapalina, i get the same prrompt you do below.

i dunno how you got that qr code prompt but it seems to me that you changed something?

**Comment by Stecca6663** [comment]
Well I just noticed the screenshots are flipped. I meant that the QR code one is chromium legacy, while the other one is Firefox.

**Comment by i3roly** [comment]
i don't think firefox supports QR code passkeys at this time, you will have to find out.

i know that the official firefox on 10.15 offered the same prompt as the one you're seeing on the second picture.

**Comment by RJVB** [comment]
On a somewhat related, positive note: at least the 1st primary password prompt tended to be quirky with the backported Waterfox-G3 (I'd have to cancel it because it wouldn't take any input). I haven't yet noticed anything like that with FF Dynasty.

**Comment by i3roly** [comment]
so can someone close this if they can confirm that firefox doesn't offer QR-based passkeys?

**Comment by Stecca6663** [comment]
I’ll confirm with a screenshot in an hour or so, but Firefox on Sonoma on my MacBook Air does show the QR code prompt

**Comment by RJVB** [comment]
Since when and on what platforms should support for passkeys be available? I'm not finding any mention of them in the settings on Linux (I'm still on a 133 beta though).

**Comment by i3roly** [comment]
i don't think he fully knows how he got passkey with qr codes because nothing in my search says it's supported out of hte box.

**Comment by Stecca6663** [comment]
https://support.mozilla.org/it/kb/compilazione-automatica-credenziali-firefox

I can’t find it in English (I’m Italian) but please read it with a translator! It says it at the end of the page more or less. I swear I’m not imagining things o.o

**Comment by i3roly** [comment]
nowhere on that page does it say QR codes are supported. it says PASSKEY is supported, through a DEVICE. not QR codes. that page exists in english as well, and it behaves exactly as the picture you showed.

you're doing something else like 2fa passwords plus passkey or something like that.

**Comment by Stecca6663** [comment]
I’m sorry I don’t want to be stubborn, but in the article it shows that there are two options, one with a device and one with a QR code. That’s what also happens on my MacBook Air. I’ll send a video when as soon as I can. I don’t have any extensions or anything of that sort. Anyways it might also be OS dependent, which in that case there is not much to do, I was just wondering why on chromium legacy it works and not on Firefox!
![IMG_3617](https://github.com/user-attachments/assets/acc59320-f922-48d2-ae98-37815ceb3b73)

**Comment by i3roly** [comment]
"Note: The option to use a Passkey will only be available on websites that support this technology. The process to use Passkeys in Firefox may vary slightly based on the version of macOS you are using."

i'll just wait for you to show me a picture of qr-code based passkeys on mac so i know where to look. that picture you're showing isn't helpful at all. i read that link. the prompt offers passkey for a device with a camera and it sais "iphone, ipad, or android device"

stop comparing my shit to chromium legacy, too. i am not in a competition with him and i don't care about chromium.

**Comment by i3roly** [comment]
i don't think that has any relevance here.

as i stated earlier: i am using the official firefox on 10.15 and i get the same prompt he showed above on my machine. so it has nothing to do with signature

**Comment by RJVB** [comment]
10.15 in a VM, no? I wouldn't be surprised if the functionality (e.g. https://developer.apple.com/documentation/authenticationservices/supporting-passkeys) depends on a hardware component.

FWIW, that document and https://searchfox.org/mozilla-central/source/dom/webauthn/MacOSWebAuthnService.mm suggest that from OS 13.3 onwards there are `some options that aren't directly settable using the public API`.


> the prompt offers passkey for a device with a camera and it sais "iphone, ipad, or android device"

Erm, not to play devil's advocate, but isn't that how QR passkeys work? You use your phone's camera to scan the code displayed on the computer monitor and then somehow phone and computer shake hands and do a merry dance?

Anyway, I have no horses in this race, I don't want to need to use my phone to unlock anything on my computers and don't see a reason for it either.

**Comment by i3roly** [comment]
no. 10.15 natively. why would i use 10.5 in a vm? it's terrible.

i am using 10.15 because it's the lowest OS supported by my laptop.

and that prompt he shared above is not in the ff source. it's from a website. i don't know which.

i don't know anythign about passkey authentication. i'm just saying that the official build behaves on 10.15 as this guy is showing in the picture. that's all.

i don't know about newer versions of mac, and quite frankly: if i did then i wouldn't have done this.

**Comment by RJVB** [comment]
On Sunday December 08 2024 14:29:23 gagan sidhu wrote:
>no. 10.15 natively. why would i use 10.5 in a vm? it's terrible.
>i am using 10.15 because it's the lowest OS supported by my laptop.

I must have misunderstood then, I thought you said on the mic issue that you couldn't compare the Dynasty and an official build on real Apple hardware new enough to run 10.15 .

>i don't know about newer versions of mac, and quite frankly: if i did then i wouldn't have done this. 

Amen to that I suppose.

Either way I don't see how it could make sense that you could use some external hardware solution with an older OS but that it'd take a newer OS to use QR codes. As far as I'm concerned you shouldn't even need platform support for this feature. You can sign into the whatsapp or telegram web clients just fine with a QR code. Seems to me that using your phone as a passkey is useful only when you keep your saved passwords in the Mozilla cloud, but in that case the QR code can be generated remotely by the FireFox accounts server. Anway, as I said, no horse in this race.

**Comment by i3roly** [comment]
well he hasn't shown firefox with qr code support. it could be related to entitlements and such, and if that's the case then it couldn't be ported backwards.

**Comment by Stecca6663** [comment]
I'm sorry, I didn't mean to compare it as a way like "see ... it's better" but in a way to show that the feature worked there. Anyway it's definitely not important and I just made this issue to see if it was possible to implement, but it's definitely not needed. I want to stress that the browser in the state it is it's great, and it's fantastic work. Anyway here's a video of what I mean on my modern machine. 
https://github.com/user-attachments/assets/f2ea1e33-9ca7-4ed7-b49d-758a8e264ad7

**Comment by i3roly** [comment]
looks like that's a feature on newer macs. not sure if it can be backported

**Comment by Stecca6663** [comment]
Yeah that's ok! You can close the issue if you want, It's really really niche. And sorry for wasting your time on this anyways...

**Comment by i3roly** [comment]
nah i'll leave it open. pretty interesting that bluebox got the functionality working. would have to effectively load the whole new framework statically for older macs.

**Comment by RJVB** [comment]
On Sunday December 08 2024 15:35:04 gagan sidhu wrote:
> would have to effectively load the whole new framework statically for older macs.

Is that possible with system frameworks, seems it would break the whole idea of OS upgrades?

**Comment by i3roly** [comment]
I don't think it is. So maybe the chrome browser is somehow using QR separately as a form of passkey whereas firefox relies on the OS api.                                                                                                                                                                                                                                                                                                                                         Sent from my BlackBerry 10 smartphone on the Rogers network.                                                                                                                                                                                                                From: René BertinSent: Sunday, 8 December 2024 5:04 PMTo: i3roly/firefox-dynastyReply To: i3roly/firefox-dynastyCc: gagan sidhu; CommentSubject: Re: [i3roly/firefox-dynasty] Passkey support incomplete on MacOS 10.9.5 (Issue #24)
On Sunday December 08 2024 15:35:04 gagan sidhu wrote:
> would have to effectively load the whole new framework statically for older macs.

Is that possible with system frameworks, seems it would break the whole idea of OS upgrades?


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/i3roly/firefox-dynasty/issues/24#issuecomment-2526524350",
"url": "https://github.com/i3roly/firefox-dynasty/issues/24#issuecomment-2526524350",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by Stecca6663** [comment]
It’s probably not useful since it’s about modern MacOS versions, but it seems like Zen Browser currently has the same issue.
https://github.com/zen-browser/desktop/issues/437

**Comment by i3roly** [comment]
those people are reporting the tap doesn't work. i don't think that's the case here.

i think you observed behaviour that exists for, at least, "macOS" 12 and higher, but it may exist in Big Turd as well.

it's not in crapalina, since the official build shows me exactly what you noticed above.

**Comment by Wowfunhappy** [comment]
I haven't tried it, because I'm mostly against passkeys (too difficult/risky to move between devices), but I believe you could probably use Bitwarden for passkey support as a workaround. (This also helps partially with the aforementioned "too risky to move between devices" problem.)

**Comment by mrrrcel76** [comment]
@i3roly unrelated to anything, just wanted to say u cooking man im thankful for this project
